test/lib/publish.js

```diff
@@ -506,3 +506,44 @@ t.test('read registry only from publishConfig', t => {
     t.end()
   })
 })
+
+t.test('able to publish after if encountered multiple configs', t => {
+  t.plan(3)
+
+  const registry = 'https://some.registry'
+  const tag = 'better-tag'
+  const publishConfig = { registry }
+  const testDir = t.testdir({
+    'package.json': JSON.stringify({
+      name: 'my-cool-pkg',
+      version: '1.0.0',
+      publishConfig,
+    }, null, 2),
+  })
+
+  const configList = [{ tag }, { registry: `https://other.registry`, tag: 'some-tag' }, defaults]
+
+  const Publish = requireInject('../../lib/publish.js', {
+    libnpmpublish: {
+      publish: (manifest, tarData, opts) => {
+        t.same(opts.defaultTag, tag, 'gets option for expected tag')
```

Comment:
I also added this assertion to check whether we're applying the config in correct priority